infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/project/OpenShiftProject.java

```diff
@@ -133,15 +134,12 @@ void deleteIfManaged() throws InfrastructureException {
 
     OpenShiftClient osClient = clientFactory.createOC(workspaceId);
 
-    if (!isManagedInternal(osClient)) {
-      throw new InfrastructureException(
-          format(
-              "Can't delete project '%s' that contains"
-                  + " runtime of workspace '%s' because it doesn't have the '"
-                  + MANAGED_NAMESPACE_LABEL
-                  + "' label equal to 'true'.",
-              projectName,
-              workspaceId));
+    if (!isProjectManaged(osClient)) {
+      LOG.debug(
+          "Project {} for workspace {} is not marked as managed. Ignoring the delete request.",
+          projectName,
+          workspaceId);
+      return;
     }
 
     delete(projectName, osClient);
```

Comment:
why you don't catch exception here, but you catch it in k8s class ?

Reply:
Let's solve this under https://github.com/eclipse/che/issues/15250